state/execution.go

```diff
@@ -417,15 +425,28 @@ func (blockExec *BlockExecutor) Commit(
 	)
 
 	// Update mempool.
-	err = blockExec.mempool.Update(
-		block.Height,
-		block.Txs,
-		abciResponse.TxResults,
-		TxPreCheck(state),
-		TxPostCheck(state),
-	)
+	memUpdate := mempoolUpdate{block: block, state: state.Copy(), abciResponse: abciResponse}
+	go blockExec.asyncUpdateMempool(unlockMempool, memUpdate)
```

Comment:
Wouldn't it be simpler to just pass all the arguments to `asyncUpdateMempool` directly (and so, we wouldn't need to define `mempoolUpdate` struct`.

Bear in mind that the actual parameters passed to `asyncUpdateMempool` are evaluated in the [caller's goroutine](https://go.dev/tour/concurrency/1).

Reply:
Great point, done!